### PR TITLE
Fixed #30505 -- Doc'd how changes in the order of Field.choices affect migrations.

### PR DESCRIPTION
**Pull request by caioariede**

### CODE REVIEW

File: docs/topics/db/models.txt [hunk 1]
```diff
@@ -154,7 +154,7 @@ ones:
     <Field.blank>`, the field will be required.
 
 :attr:`~Field.choices`
-    An :term:`iterable` of 2-tuples to use as choices for this field. If this
+    A :term:`sequence` of 2-tuples to use as choices for this field. If this
```
Comment (felixxm) [inline docs/topics/db/models.txt]: Looks good, but I think that is just not enough for an issue described in the ticket, i.e.
```python
ACTIVITY_TYPES = set(['Create', 'Update', 'Delete'])
...
field = models.CharField(
     choices=[(x,x) for x in ACTIVITY_TYPES],
     ...
```
in this case `choices` is a list but an order is changing. I would add a note:
```
.. note::
    A new migration is created each time an order of ``choices`` changes.
```

**Comment by caioariede** [comment]
Screen Shot 2019-06-04 at 8 04 13 AM

--- 

Screen Shot 2019-06-04 at 8 04 24 AM

File: docs/ref/models/fields.txt [hunk 2]
```diff
@@ -80,12 +80,15 @@ If a field has ``blank=False``, the field will be required.
...
 
-An :term:`iterable` consisting itself of iterables of exactly two items (e.g.
+A :term:`sequence` consisting itself of iterables of exactly two items (e.g.
 ``[(A, B), (A, B) ...]``) to use as choices for this field. If choices are
 given, they're enforced by :ref:`model validation <validating-objects>` and the
 default form widget will be a select box with these choices instead of the
 standard text field.
 
+.. note::
+    A new migration is created each time the order of ``choices`` changes.
```
Comment (felixxm) [inline docs/ref/models/fields.txt]: I found this note more useful after paragraph about "dynamic choices" .

**Comment by felixxm** [comment]
Can you allow me (maintainers) to push changes into your branch?

**Comment by caioariede** [comment]
Yeah @felixxm

**Comment by felixxm** [comment]
@caioariede Thanks!

**Pull request merged**